src/Illuminate/Queue/CallQueuedHandler.php

```diff
@@ -66,7 +66,7 @@ public function call(Job $job, array $data)
      */
     protected function dispatchThroughMiddleware(Job $job, $command)
     {
-        return (new Pipeline)->send($command)
+        return app(Pipeline::class)->send($command)
```

Comment:
`app` helper only exists in Foundation that belongs to the whole framework and hasn't been released while `illuminate/queue` can be considered as a standalone component. So, I think we have to use `Container::getInstance` instead.

```php
return new Pipeline(Container::getInstance())->send($command)
```

Reply:
@DarkGhostHunter You can inject the container interface via the constructor so that we do not depend here on the illuminate concrete implementation.

Reply:
I could have, but this was already merged lol.

The CallQueuedListener instancing happens in the [`Illuminate/Events/Dispatcher`](https://github.com/laravel/framework/blob/5.8/src/Illuminate/Events/Dispatcher.php#L483-L490) where is just a normal instancing. Here it could inject the IoC of the Dispatcher, since it uses the implementation instead of a concrete.

Don't know what @taylorotwell thinks though, since it has been merged I assume this is an "OK" from him.

Reply:
Sent in a followup PR: https://github.com/laravel/framework/pull/29547